### ISSUE TITLE
After updating from sdk45 to 48, in prod build assets don't work #22914 #22916

### ISSUE THREAD

**Issue by evgenyshenets91** [expo/expo#22918]
### Minimal reproducible example

<Image         style={{ width: props.size || 26, height: props.size || 26 }}         source={require('../../../assets/animated/spinner.gif')}       />

### Summary

After updating from expo sdk 45 to 48, found out that on the prod build Assets don't appear. On dev build works correctly. Found out that flipper return to me an error CUICatalog: Invalid asset name supplied: '(null)'. After extracting the .ipa file found out that in the assets folder all images exist.

### Environment

Environment
expo-env-info 1.0.5 environment info:
System:
OS: macOS 13.0.1
Shell: 5.8.1 - /bin/zsh
Binaries:
Node: 18.15.0 - ~/.nvm/versions/node/v18.15.0/bin/node
Yarn: 1.22.10 - /usr/local/bin/yarn
npm: 9.5.0 - ~/.nvm/versions/node/v18.15.0/bin/npm
Watchman: 2023.05.22.00 - /usr/local/bin/watchman
Managers:
CocoaPods: 1.11.3 - /usr/local/bin/pod
SDKs:
iOS SDK:
Platforms: DriverKit 22.1, iOS 16.1, macOS 13.0, tvOS 16.1, watchOS 9.1
Android SDK:
API Levels: 23, 28, 29, 30, 31, 32, 33
Build Tools: 28.0.3, 29.0.2, 29.0.3, 30.0.2, 30.0.3, 31.0.0, 32.0.0
System Images: android-29 | Intel x86 Atom_64, android-29 | Google APIs Intel x86 Atom, android-29 | Google Play Intel x86 Atom, android-30 | Google APIs Intel x86 Atom, android-30 | Google Play Intel x86 Atom, android-31 | Google APIs ARM 64 v8a, android-31 | Google APIs Intel x86 Atom_64, android-32 | Google APIs Intel x86_64 Atom
IDEs:
Android Studio: 2021.1 AI-211.7628.21.2111.8309675
Xcode: 14.1/14B47b - /usr/bin/xcodebuild
npmPackages:
@expo/metro-config: ^0.7.1 => 0.7.1
@expo/webpack-config: ^18.0.1 => 18.0.4
expo: ~48.0.15 => 48.0.15
react: 18.2.0 => 18.2.0
react-dom: 18.2.0 => 18.2.0
react-native: 0.71.7 => 0.71.7
react-native-web: ~0.18.11 => 0.18.12
Expo Workflow: managed

**Comment by sregg** [comment]
We're seeing similar issue here: https://github.com/expo/expo/issues/22656
@evgenyshenets91 did you find a fix?

**Comment by evgenyshenets** [comment]
> We're seeing similar issue here: https://github.com/expo/expo/issues/22656
> 
> @evgenyshenets91 did you find a fix?

Yes. It was missing deals in metro config. It you extend config, don't forget ...rest all properties. 
![image](https://github.com/expo/expo/assets/102376591/844e5476-f539-4212-81ca-1d751c07cca3)

Don't remember proper plugin, but an issue was there.

**Comment by sregg** [comment]
Thanks.
I updated my config file but it's still not working:
```javascript

const { getDefaultConfig } = require('expo/metro-config');

module.exports = (async () => {
  const config = await getDefaultConfig(__dirname);
  return {
    ...config,
    transformer: {
      ...config.transformer,
      getTransformOptions: async () => ({
        transform: {
          experimentalImportSupport: false,
          inlineRequires: true,
        },
      }),
      babelTransformerPath: require.resolve('react-native-svg-transformer'),
    },
    resolver: {
      ...config.resolver,
      assetExts: config.resolver.assetExts.filter((ext) => ext !== 'svg'),
      sourceExts: [...config.resolver.sourceExts, 'svg'],
      // this is needed for Storybook (see https://github.com/storybookjs/react-native/issues/405#issuecomment-1436683333)
      resolverMainFields: ['sbmodern', ...config.resolver.resolverMainFields],
    },
  };
})();
```

**Comment by evgenyshenets** [comment]
> Thanks.
> 
> I updated my config file but it's still not working:
> 
> ```javascript
> 
> 
> 
> const { getDefaultConfig } = require('expo/metro-config');
> 
> 
> 
> module.exports = (async () => {
> 
>   const config = await getDefaultConfig(__dirname);
> 
>   return {
> 
>     ...config,
> 
>     transformer: {
> 
>       ...config.transformer,
> 
>       getTransformOptions: async () => ({
> 
>         transform: {
> 
>           experimentalImportSupport: false,
> 
>           inlineRequires: true,
> 
>         },
> 
>       }),
> 
>       babelTransformerPath: require.resolve('react-native-svg-transformer'),
> 
>     },
> 
>     resolver: {
> 
>       ...config.resolver,
> 
>       assetExts: config.resolver.assetExts.filter((ext) => ext !== 'svg'),
> 
>       sourceExts: [...config.resolver.sourceExts, 'svg'],
> 
>       // this is needed for Storybook (see https://github.com/storybookjs/react-native/issues/405#issuecomment-1436683333)
> 
>       resolverMainFields: ['sbmodern', ...config.resolver.resolverMainFields],
> 
>     },
> 
>   };
> 
> })();
> 
> ```

So, how I found out an issue in my case: I downloaded the build to laptop and then extracted ipa file. After that I found out that in assets folder I have my images, etc, but they had different naming, like random numbers and letters, eventually I understood that sh with plugin. You can begin from same way, extract build the check how assets looks like, maybe they even not in folder... good luck

**Comment by sregg** [comment]
Mine don't seem to be renamed at all
<img width="1552" alt="image" src="https://github.com/expo/expo/assets/1479283/d5c3acc3-dde5-4639-b971-460c380f0973">

**Comment by evgenyshenets** [comment]
That's really strange and means that the problem not in plugin for transforming. So you can also add in your image component props onError, add alert and have a look what error will return prod build.

**Comment by sregg** [comment]
I was missing `import 'expo-asset';` in my app.
https://github.com/expo/expo/issues/22656#issuecomment-1629401431

**Comment by evgenyshenets** [comment]
Glad to hear!
